### PR TITLE
*: add roadmap

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

add roadmap

### Check List 

Tests 

 - No code

### CODE REVIEW

File: roadmap.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- Make DM easy to use: there are a lot of usability issues that have been spat out for up to a year! We need to address at least known usability issues.
+- DM 2.0 GA: DM 2.0 (supports High Availability and better shard DDL replication) has taken so long and no GA, it’s time to GA delivery.
+
+## Usability Improvement
+
+- [ ] solve known usability issue (continuous work)
+  - What: solve usability issues recorded in [the project](https://github.com/pingcap/dm/projects/3)
+  - Why: a lot of usability issues that have been not resolved yet, we need to stop user churn
+- [ ] better configuration file [#775](https://github.com/pingcap/dm/issues/775)
+  - What: avoid misuse for configuration files
```
Comment (lance6716) [inline roadmap.md:14]:
```suggestion
  - What: avoid misusing for configuration files
```

File: roadmap.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  - Why: many users meet problem when write configuration file but don’t know how to deal with it
+
+## New features
+
+- [ ] stop/pause until reached the end of a transaction [#1095](https://github.com/pingcap/dm/issues/1095)
+  - What: replicate binlog events until reached the end of the current transaction when stopping or pausing the task
+  - Why: achieve transaction consistency as the upstream MySQL after stopped/paused the task
+- [ ] stop at the specified position/GTID [#348](https://github.com/pingcap/dm/issues/348)
+  - What: stop to replicate when reached the specified binlog position or GTID, like `until_option` in MySQL
+  - Why: no more data replicated without needs to stop writing in the upstream
```
Comment (lance6716) [inline roadmap.md:24]: confusing sentence 🤔

File: roadmap.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+- [ ] update source config online [#1076](https://github.com/pingcap/dm/issues/1076)
+  - What: update source configs (like upstream connection arguments) online
+  - Why: switch from one MySQL instance to another in the replica group easier
+- [ ] provides a complete online replication checksum feature [#1097](https://github.com/pingcap/dm/issues/1097)
+  - What:
+    - check data without stopping writing in upstream
+    - no extra writes  in upstream
```
Comment (lance6716) [inline roadmap.md:31]:
```suggestion
    - no extra writes in upstream
```

File: roadmap.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  - Why: found potential consistency earlier
```
Comment (lance6716) [inline roadmap.md:32]:
```suggestion
  - Why: found potential inconsistency earlier
```

File: roadmap.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+- [ ] support DM v2.0 in TiDB Operator [tidb-operator#2868](https://github.com/pingcap/tidb-operator/issues/2868)
+  - What: use TiDB-Operator to manage DM 2.0
+
+## Out of Scope currently
+
+- Only supports synchronization of MySQL protocol Binlog to TiDB cluster and not all MySQL protocol databases
```
Comment (lance6716) [inline roadmap.md:38]:
```suggestion
- Only supports synchronization of MySQL protocol Binlog to TiDB cluster but not all MySQL protocol databases
```

File: roadmap.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  - Why: some MySQL protocol databases are not binlog protocol compatible
+- Provides a fully automated shard DDL merge and synchronization solution
+  - Why: many scenes are difficult to automate
+- Replicate multiple upstream MySQL sources with one dm-worker
```
Comment (lance6716) [inline roadmap.md:42]:
```suggestion
- Replicates multiple upstream MySQL sources with one dm-worker
```

File: roadmap.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  - Why:lLarge amount of development work and many uncertainties
```
Comment (lance6716) [inline roadmap.md:43]:
```suggestion
  - Why: Large amount of development work and many uncertainties
```

File: roadmap.md (hunk 2 above)
Comment (csuzhangxc) [reply roadmap.md]: rewrite in https://github.com/pingcap/dm/pull/1098/commits/1e7a3f7d89a480f9bc4187c615fa2a88ffd83f14. but may still be confused, 😢

**Review by lance6716 — COMMENTED** [review]
LGTM

**Review by IANTHEREAL — APPROVED** [review]

**Pull request merged**